jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1901,72 +1900,75 @@ else if (scheme.equalsIgnoreCase("file"))
     }
 
     /**
-     * <p>Convert a String into a URI suitable for use as a Resource.</p>
+     * <p>Convert a String reference into an absolute URI.</p>
      *
-     * @param resource If the string starts with one of the ALLOWED_SCHEMES, then it is assumed to be a
-     * representation of a {@link URI}, otherwise it is treated as a {@link Path}.
-     * @return The {@link URI} form of the resource.
-     * @deprecated This method is currently resolving relative paths against the current directory, which is a mechanism
-     * that should be implemented by a {@link ResourceFactory}.   All calls to this method need to be reviewed.
+     * <p>If given a relative reference string to a path, it will convert it to an absolute path
+     * using the same techniques present in the JVM itself (eg: {@code Path.of(reference).toAbsolutePath()})</p>
+     *
+     * <p>Relative URI reference strings (eg: {@code "file:path/dir/"}
+     * and {@code "file:path/foo.jar!/"}) are not supported.</p>
+     *
+     * @param reference the raw String input.
+     * @return The absolute {@link URI} form of the reference.
+     * @throws IllegalArgumentException if unable to convert the input string.
      */
-    @Deprecated(since = "12.0.8")
-    public static URI toURI(String resource)
+    public static URI toURI(String reference)
     {
-        Objects.requireNonNull(resource);
+        Objects.requireNonNull(reference);
 
-        if (URIUtil.hasScheme(resource))
+        try
         {
-            try
+            /* Perform URI test first.
+             * We don't want to perform Path.of(String) first.
+             *
+             * Example: reference parameter is the String "file:///path/to/dir"
+             *
+             * On Unix, the Path.of(reference) will result in a relative directory reference
+             * that includes the "file:" portion in the path after the current working directory.
+             * You'll wind up with something like "file:///home/user/code/jetty/12.1.x/jetty-core/jetty-util/file:///path/to/dir" in this case
+             *
+             * On Windows, the Path.of(reference) will not allow a Path.of("file:///path/to/dir") to work.
+             * This is because there cannot be multi-character drive letters (yes, Windows is limited to only 26 drive letters max)
+             */
+            URI uri = URI.create(reference);
+            if (uri.isAbsolute())
             {
-                URI uri = new URI(resource);
-
-                if (ResourceFactory.isSupported(uri))
-                    return correctURI(uri);
-
-                // We don't have a supported URI scheme
-                if (uri.getScheme().length() == 1)
+                // At this point we have a string detected as a URI.
+                // But that could also include Windows paths like "C:\path\to\foo.jar" or "C:/path/to/foo.jar"
+                String scheme = uri.getScheme();
```

Comment:
If on windows I pass in a `reference` of `file:///c:/some/thing.txt` then is the `scheme` reported after conversion to `URI` `file` or `c`?

Reply:
See the testcases in ResourceFactoryTest.testSplitIsAbsolute for some windows examples.